channels/postgres/channels.go

```diff
@@ -424,6 +424,12 @@ func (cr *channelRepository) RetrieveAll(ctx context.Context, pm channels.Page)
 					%s
 					`, connJoinQuery, pageQuery)
 
+	cq := fmt.Sprintf(`SELECT COUNT(*) AS total_count
+				FROM (
+					%s
```

Comment:
@nyagamunene 
I appreciate your changes, which resulted in a smaller diff. However, this approach duplicates the total calculation function. What do you think about the following alternative approach?? 

```suggestion
	var items []channels.Channel
	if !pm.OnlyTotal {
		rows, err := cr.db.NamedQueryContext(ctx, q, dbPage)
		if err != nil {
			return channels.ChannelsPage{}, errors.Wrap(repoerr.ErrFailedToRetrieveAllGroups, err)
		}
		defer rows.Close()

		for rows.Next() {
			dbch := dbChannel{}
			if err := rows.StructScan(&dbch); err != nil {
				return channels.ChannelsPage{}, errors.Wrap(repoerr.ErrViewEntity, err)
			}

			ch, err := toChannel(dbch)
			if err != nil {
				return channels.ChannelsPage{}, err
			}

			items = append(items, ch)
		}
	}
```